### PR TITLE
improved hex debug print, improved USBH debug prints, added pid/vid debug print

### PR DESCRIPTION
**Pull request by IngHK**
- improved debug print of hex values TU_LOG1_HEX, added "0x" before hex value print (src/common/tusb_debug.h)
- added macros for int, hex, mem and buf debug prints in src/host/usbh_pvt.h
- added pid and vid print in src/host/usbh.c
- clean up debug prints in src/host/usbh.c

### CODE REVIEW

File: src/host/usbh.c [hunk 1]
```diff
@@ -291,6 +291,9 @@ bool tuh_vid_pid_get(uint8_t dev_addr, uint16_t *vid, uint16_t *pid) {
   *vid = dev->vid;
   *pid = dev->pid;
 
+  TU_LOG_HEX_USBH(*vid);
```
Comment (hathach) [inline src/host/usbh.c:294]: please remove these 2 log.

File: src/host/usbh.c [hunk 2]
```diff
@@ -860,7 +863,7 @@ bool usbh_edpt_xfer_with_callback(uint8_t dev_addr, uint8_t ep_addr, uint8_t * b
     // HCD error, mark endpoint as ready to allow next transfer
     ep_state->busy    = 0;
     ep_state->claimed = 0;
-    TU_LOG1("Failed\r\n");
+    TU_LOG_USBH("Failed\r\n");
```
Comment (hathach) [inline src/host/usbh.c:866]: keep this as LOG1, since failed is always print out with LOG LEVEL = 1

File: src/host/usbh.c [hunk 3]
```diff
@@ -1318,7 +1321,7 @@ static void process_enumeration(tuh_xfer_t* xfer) {
     if ( retry ) {
       failed_count++;
       osal_task_delay(ATTEMPT_DELAY_MS); // delay a bit
-      TU_LOG1("Enumeration attempt %u\r\n", failed_count);
+      TU_LOG_USBH("Enumeration attempt %u\r\n", failed_count);
```
Comment (hathach) [inline src/host/usbh.c:1324]: please keep LOG1, consider as failed attempt

**Review by hathach — CHANGES_REQUESTED** [review]
look great, thank you, only a few change request per review

File: src/host/usbh.c (hunk 1 above)
Comment (IngHK) [reply src/host/usbh.c]: done

File: src/host/usbh.c (hunk 2 above)
Comment (IngHK) [reply src/host/usbh.c]: done

File: src/host/usbh.c (hunk 3 above)
Comment (IngHK) [reply src/host/usbh.c]: done

**Pull request merged**